custom-remote-user-store-manager/manager/src/main/java/org/wso2/carbon/remote/user/store/manager/WSUserStoreManager.java

```diff
@@ -0,0 +1,388 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * User store manager for the web service user store manager.
+ */
+public class WSUserStoreManager extends AbstractUserStoreManager {
+
+    private static final String WS_UM_PROPERTY_URL_NAME = "url";
+    private static final String WS_UM_PROPERTY_URL_DISPLAY_NAME = "http://localhost:8080/sample-service";
+    private static final String WS_UM_PROPERTY_URL_DESCRIPTION = "Endpoint URL for the user store web service";
+    private static Log log = LogFactory.getLog(WSUserStoreManager.class);
+    private CloseableHttpClient closeableHttpClient;
+
+    public WSUserStoreManager() {
+
+    }
+
+    public WSUserStoreManager(RealmConfiguration realmConfig, Map<String, Object> properties,
+                              ClaimManager claimManager, ProfileConfigurationManager profileManager, UserRealm realm,
+                              Integer tenantId) throws UserStoreException {
+
+        this.realmConfig = realmConfig;
+        this.claimManager = claimManager;
+        this.userRealm = realm;
+        this.tenantId = tenantId;
+        realmConfig.setUserStoreProperties(JDBCRealmUtil.getSQL(realmConfig
+                .getUserStoreProperties()));
+
+        // new properties after carbon core 4.0.7 release.
+        if (realmConfig.getUserStoreProperty(UserCoreConstants.RealmConfig.READ_GROUPS_ENABLED) != null) {
+            readGroupsEnabled = Boolean.parseBoolean(realmConfig
+                    .getUserStoreProperty(UserCoreConstants.RealmConfig.READ_GROUPS_ENABLED));
+        }
+
+        if (log.isDebugEnabled()) {
+            if (readGroupsEnabled) {
+                log.debug("ReadGroups is enabled for " + getMyDomainName());
+            } else {
+                log.debug("ReadGroups is disabled for " + getMyDomainName());
+            }
+        }
+
+        if (realmConfig.getUserStoreProperty(UserCoreConstants.RealmConfig.WRITE_GROUPS_ENABLED) != null) {
+            writeGroupsEnabled = Boolean.parseBoolean(realmConfig
+                    .getUserStoreProperty(UserCoreConstants.RealmConfig.WRITE_GROUPS_ENABLED));
+        } else {
+            if (!isReadOnly()) {
+                writeGroupsEnabled = true;
+            }
+        }
+
+        if (log.isDebugEnabled()) {
+            if (writeGroupsEnabled) {
+                log.debug("WriteGroups is enabled for " + getMyDomainName());
+            } else {
+                log.debug("WriteGroups is disabled for " + getMyDomainName());
+            }
+        }
+
+        // This property is now deprecated
+        if (realmConfig
+                .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_INTERNAL_ROLES_ONLY) != null) {
+            boolean internalRolesOnly = Boolean
+                    .parseBoolean(realmConfig
+                            .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_INTERNAL_ROLES_ONLY));
```

Comment:
Can you please format this line correctly?

Reply:
Fixed in 09486ce0c6187db78c1b97e4eaefd890c511141b.